src/main/controllers/WorkAddressController.ts

```diff
@@ -34,10 +34,12 @@ export default class WorkAddressController {
   }
 
   public post = (req: AppRequest, res: Response): void => {
+    setUserCase(req, this.form);
     const { saveForLater } = req.body;
 
     if (saveForLater) {
-      handleSaveAsDraft(res);
+      handleSessionErrors(req, res, this.form, PageUrls.CLAIM_SAVED);
+      handleUpdateDraftCase(req, this.logger);
```

Comment:
This line is the same in both branches. Is it worth moving this outside of the if statement entirely?